src/git/mod.rs

```diff
@@ -71,64 +64,75 @@ impl<'cb> RepoCloneBuilder<'cb> {
         }
 
         if let Some(identity_path) = identity_path {
-            builder.set_identity(identity_path);
+            builder.set_identity(identity_path)?;
         }
 
         Ok(builder)
     }
 
-    pub fn set_identity(&mut self, identity_path: &Path) {
-        self.identity = Some(PathBuf::from(identity_path));
+    pub fn set_identity(&mut self, identity_path: &Path) -> Result<()> {
+        let identity_path = utils::canonicalize_path(identity_path)?;
+        log::info!(
+            "{} `{}` {}",
+            style("Using private key:").bold(),
+            style(format_args!("{}", identity_path.display()))
+                .bold()
+                .yellow(),
+            style("for git-ssh checkout").bold()
+        );
+        self.authenticator =
+            GitAuthenticator::new_empty().add_ssh_key_from_file(identity_path, None);
```

Comment:
```suggestion
            GitAuthenticator::new_empty().add_ssh_key_from_file(identity_path, None)            .prompt_ssh_key_password(true);
```

This change would support passphrase protected ssh key files, because it would still ask for the passphrase if needed.

Reply:
Ah, good point. I forgot about enabling that :)